src/test/java/edu/cornell/kfs/sys/service/mock/MockAwsSecretServiceImplTest.java

```diff
@@ -0,0 +1,154 @@
+package edu.cornell.kfs.sys.service.mock;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+
+import edu.cornell.kfs.sys.CuSysTestConstants.MockAwsSecretServiceConstants;
+import edu.cornell.kfs.sys.service.impl.fixture.AwsSecretPojo;
+import edu.cornell.kfs.sys.util.CuJsonUtils;
+
+/**
+ * This is a modified copy of the cu-kfs AwsSecretServiceImplIntegrationTest class,
+ * which has been changed to validate the Mock AWS Secret Service instead.
+ */
+public class MockAwsSecretServiceImplTest {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final String SINGLE_STRING_SECRET_KEY_NAME = "unittest/singlestring";
+    private static final String SINGLE_DATE_SECRET_KEY_NAME = "unittest/singledate";
+    private static final String BASIC_POJO_SECRET_KEY_NAME = "unittest/pojo";
+    private static final String SINGLE_BOOLEAN_SECRET_KEY_NAME = "unittest/singleboolean";
+    private static final String SINGLE_FLOAT_SECRET_KEY_NAME = "unittest/singlefloat";
+    
+    private static final String STATIC_STRING_PROPERTY_NAME = "static_string";
+    private static final String NUMBER_TEST_PROPERTY_NAME = "number_test";
+    
+    private static final String SINGLE_STRING_SECRET_VALUE = "Test Value";
+    private static final String BASIC_POJO_STATIC_STRING_VALUE = "do not change me";
+    private static final int BASIC_POJO_NUMBER_VALUE = 1;
+    
+    private MockAwsSecretServiceImpl awsSecretServiceImpl;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        awsSecretServiceImpl = new MockAwsSecretServiceImpl();
+        awsSecretServiceImpl.setAwsRegion(MockAwsSecretServiceConstants.AWS_US_EAST_ONE_REGION);
```

Comment:
As we move more functionality into AWS Secrets Manager, I expect MockAwsSecretServiceImpl will be used more often.  Do you think setting the region, namespaces and retry count could be moved into the constructor to make reuse easier?

Reply:
Ok, I'll add more constructor options to make the mock secret service easier to set up.